### PR TITLE
[ROX-10969] : Change Vuln reporting to use imageComponents and imageVulnerabilities graphQL resolvers

### PR DESCRIPTION
**Pull request by charmik-redhat**
## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

Manual testing to check if vuln report is correctly generated

4DEEEDF0-050A-4515-8818-626FA73E9C06_1_105_c

### CODE REVIEW

**Review by md2119 — APPROVED** [review]
minor comments (technically not related to your PR)

**Review by keyallis — APPROVED** [review]
I'd recommend taking Mandar's recommendations on naming and then merging, looks good

File: central/reports/common/report_formatter.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ var (
 	}
 )
 
-type vulnObj struct {
+type ImageVulnerabilityObj struct {
```
[the remark this answers is not in the document]
Comment (keyallis) [reply central/reports/common/report_formatter.go:38]: This name should also start with a lowercase letter

File: central/reports/common/report_formatter.go [hunk 2]
```diff
@@ -44,14 +44,14 @@ type vulnObj struct {
 	Link              string        `json:"link,omitempty"`
 }
 
-type compObj struct {
-	Name  string     `json:"name,omitempty"`
-	Vulns []*vulnObj `json:"vulns,omitempty"`
+type ImageComponentObj struct {
```
[the remark this answers is not in the document]
Comment (keyallis) [reply central/reports/common/report_formatter.go:47]: This name should also start with a lowercase letter

**Pull request merged**